### ISSUE TITLE
DocGen fails to generate __docgen Infos in bundled code

### ISSUE THREAD

**Issue by chrkulbe** [storybookjs/storybook#15052]
**Describe the bug**
The bundled source files with webpack 5 missing the docgen information in the bundeled components.
Eg. http://localhost:6006/?path=/story/example-button--primary

webpack 4:  4 controls + onClick in the controls addon; events will be logged in the actions addon
webpack 5:  3 controls in the controls addon; events will **NOT** logged in the actions addon

**To Reproduce**
npx sb init
vs.
npx sb init --builder webpack5

and start the demo app

**System**
Environment Info:

  System:
    OS: Windows 10 10.0.19041
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
  Binaries:
    Node: 14.15.0 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.8 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.964.0), Chromium (90.0.818.66)
  npmPackages:
    @storybook/addon-a11y: ^6.2.9 => 6.2.9
    @storybook/addon-actions: ^6.2.9 => 6.2.9
    @storybook/addon-essentials: ^6.2.9 => 6.2.9
    @storybook/addon-links: ^6.2.9 => 6.2.9
    @storybook/addons: ^6.2.9 => 6.2.9
    @storybook/builder-webpack5: ^6.2.9 => 6.2.9
    @storybook/react: ^6.2.9 => 6.2.9
    @storybook/theming: ^6.2.9 => 6.2.9

**Additional context**
The try block is completely missing with webpack 5.

```javascript
try {
    // @ts-ignore
    Button.displayName = "Button";
    // @ts-ignore
    Button.__docgenInfo = { "description": "Primary UI component for user interaction", "displayName": "Button", "props": { "primary": { "defaultValue": { value: "false" }, "description": "Is this the principal call to action on the page?", "name": "primary", "required": false, "type": { "name": "boolean" } }, "backgroundColor": { "defaultValue": null, "description": "What background color to use", "name": "backgroundColor", "required": false, "type": { "name": "string" } }, "size": { "defaultValue": { value: "medium" }, "description": "How large should the button be?", "name": "size", "required": false, "type": { "name": "enum", "value": [{ "value": "undefined" }, { "value": "\"small\"" }, { "value": "\"medium\"" }, { "value": "\"large\"" }] } }, "label": { "defaultValue": null, "description": "Button contents", "name": "label", "required": true, "type": { "name": "string" } }, "onClick": { "defaultValue": null, "description": "Optional click handler", "name": "onClick", "required": false, "type": { "name": "(() => void)" } } } };
    // @ts-ignore
    if (typeof STORYBOOK_REACT_CLASSES !== "undefined")
        // @ts-ignore
        STORYBOOK_REACT_CLASSES["src/stories/Button.tsx#Button"] = { docgenInfo: Button.__docgenInfo, name: "Button", path: "src/stories/Button.tsx#Button" };
}
catch (__react_docgen_typescript_loader_error) { }
```

**Comment by WalterWeidner** [comment]
Just ran into this as well in our project. All of our descriptions in DocsTable are gone:

![image](https://user-images.githubusercontent.com/3867422/119696966-f852e800-be0c-11eb-9fdc-5d794b1389a2.png)

**Comment by shilman** [comment]
Closing as duplicate to #14118, which has been fixed in 6.3. To upgrade to the latest prerelease:

```
npx sb@next upgrade --prerelease
```
